include/boost/geometry/strategies/geographic/intersection.hpp

```diff
@@ -76,6 +77,12 @@ struct geographic_segments
 {
     typedef geographic_tag cs_tag;
 
+    inline geographic_segments<FormulaPolicy, Order, Spheroid, CalculationType>
+    get_relate_segment_segment_strategy() const
+    {
+        return geographic_segments<FormulaPolicy, Order, Spheroid, CalculationType>(m_spheroid);
+    }
```

Comment:
Why does this strategy has to define a getter to itself?

Reply:
it is the issue with cyclic dependences described above.